### ISSUE TITLE
systemd-run --user does not allow limiting memory

### ISSUE THREAD

**Issue by whatdoineed2do** [systemd/systemd#10581]
**systemd version the issue has been seen with**
systemd 238
+PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid

**Used distribution**
Fedora 28

**Expected behaviour you didn't see**
Limiting resource via `--user` should limit as long as they are lower than corresponding limits in parent slice/unit.

`systemd-run --user --unit=dev-limit.service -t -p MemoryMax=50M -p MemorySwapMax=100M -p CPUQuota=50% stress -cpu 8 -vm 2 -vm-bytes 256M`

Should be able to request a _lower_ limit (like `ulimit`)
```
● dev-limit1.service - /bin/stress --cpu 8 --vm 2 --vm-bytes 256M
   Loaded: loaded (/run/systemd/transient/dev-limit1.service; transient)
Transient: yes
   Active: active (running) since Tue 2018-10-30 20:01:47 GMT; 9s ago
 Main PID: 18071 (stress)
    Tasks: 11 (limit: 2343)
   Memory: 50.0M (max: 50.0M swap max: 100.0M)
   CGroup: /user.slice/user-500.slice/dev-limit1.service
           ├─18071 /bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─18073 /bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─18074 /bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─18075 /bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─18076 /bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─18077 /bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─18078 /bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─18079 /bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─18080 /bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─18081 /bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           └─18082 /bin/stress --cpu 8 --vm 2 --vm-bytes 256M
```

**Unexpected behaviour you saw**
Limits not obeyed.
```
● dev-limit1.service - /usr/bin/stress --cpu 8 --vm 2 --vm-bytes 256M
   Loaded: loaded (/run/user/500/systemd/transient/dev-limit1.service; transient)
Transient: yes
   Active: active (running) since Tue 2018-10-30 19:59:23 GMT; 17s ago
 Main PID: 17978 (stress)
   CGroup: /user.slice/user-500.slice/user@500.service/dev-limit1.service
           ├─17978 /usr/bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─17979 /usr/bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─17980 /usr/bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─17981 /usr/bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─17982 /usr/bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─17983 /usr/bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─17984 /usr/bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─17985 /usr/bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─17986 /usr/bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           ├─17987 /usr/bin/stress --cpu 8 --vm 2 --vm-bytes 256M
           └─17988 /usr/bin/stress --cpu 8 --vm 2 --vm-bytes 256M
```
**Steps to reproduce the problem**
This may be user error but we can only set limit (via `sudo`) at a higher scope.

Works (limiting in place):
```
sudo systemd-run -t --slice=user-$(id -u).slice --unit=dev-limit1.service -p MemoryMax=50M -p MemorySwapMax=100M -p CPUQuota=50% --uid ray stress --cpu 8 --vm 2 --vm-bytes 256M
```
Not working:
```
systemd-run --user -t --slice=user-$(id -u).slice --unit=dev-limit1.service -p MemoryMax=50M -p MemorySwapMax=100M -p CPUQuota=50%  stress --cpu 8 --vm 2 --vm-bytes 256M
```

**Comment by poettering** [comment]
delegation of controllers to unprivileged programs is not safe on cgroupsv1. That's a kernel limitation, and we hence won't do it.

On cgroupsv2 delegation of such controllers to unprivileged programs is safe, and hence we do it by default for the "pids" and "memory" controller, but only for those. This 

Resource management with controllers that are not delegated is not available. 

Fedora still defaults to cgroupsv1 mode, since Docker and all that container jumble has not been updated for cgroupsv2 mode yet.

Long story short, if you want that what you are trying to do works you have to boot your system in cgroupsv2 mode, i.e. with `systemd.unified_cgroup_hierarchy=1` on the kernel command line, but then you cannot use Docker and all that container stuff on your system.

Anyway, closing, as this is pretty much expected (though unfortunate). This will start magically work as soon as downstream (i.e. Fedora) switches to cgroupsv2.

**Comment by cytrinox** [comment]
This may be related to #9502. I've switched to cgroupsv2 but still not works as expected. Maybe the PR https://github.com/systemd/systemd/pull/10894 fix the issue.
